src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -467,13 +467,34 @@ _public_ int sd_pid_notify_with_fds(
         if (!e)
                 return 0;
 
-        r = sockaddr_un_set_path(&sockaddr.un, e);
+        /* Allow AF_UNIX and AF_VSOCK, reject the rest. */
+        r = socket_local_address_parse(&address, e);
         if (r < 0)
                 goto finish;
-        msghdr.msg_namelen = r;
+        msghdr.msg_namelen = address.size;
+
+        /* If we didn't get an address, talk to the hypervisor by default, as that's the purpose of this functionality. */
+        if (address.sockaddr.vm.svm_family == AF_VSOCK && address.sockaddr.vm.svm_cid == VMADDR_CID_ANY)
+                address.sockaddr.vm.svm_cid = VMADDR_CID_HYPERVISOR;
+
+        /* At the time of writing QEMU does not yet support AF_VSOCK + SOCK_DGRAM and returns
+         * ENODEV. Fallback to SOCK_SEQPACKET in that case. */
+        fd = socket(address.sockaddr.sa.sa_family, SOCK_DGRAM|SOCK_CLOEXEC, 0);
+        if (fd < 0 && errno == ENODEV && address.sockaddr.sa.sa_family == AF_VSOCK) {
```

Comment:
ENODEV is such a weird error. Maybe add ERRNO_IS_NOT_SUPPORTED() here? (simple to cover for seccomp filters at the very least)